src/System.ValueTuple/src/System/Numerics/Hashing/Hash.cs

```diff
@@ -0,0 +1,101 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace System.Numerics.Hashing
+{
+    internal static class Hash
```

Comment:
> if we didn't have these functions we'd have to write something like

No, we already have the CombineHashCodes methods in ValueTuple.

> which is significantly longer

But it also introduces another generic method that'll need to be compiled, and will likely need a new compilation of it for every combination of Ts that come in (assuming they're not all reference types).  Plus every item is getting passed to another method, so if any of them are big, it's yet another copy.